htslib/sam.h

```diff
@@ -1110,21 +1110,21 @@ int bam_set_qname(bam1_t *b, const char *qname);
                        can be NULL
  @param  a_cigar [out]  address of the destination uint32_t buffer
```

Comment:
a_cigar is [in/out] presumably?

Reply:
Initially, I thought of just being `[out]`, as the information contained by the buffer is generated inside the method, but the pointer itself is being read. So, I guess a case can be made for `[in]`.

Reply:
It's `in` because the pointed-to variable needs to already have a meaningful value before the call (this is how it differs from `end`, whose pointed-to variable may even be uninitialised). It's `out` because the pointed-to value might be different after the call.

Therefore it's `[in/out]`, the same as `a_mem`.